docs/orchestrating-elastic-stack-applications/elasticsearch/autoscaling.asciidoc

```diff
@@ -0,0 +1,197 @@
+
+[source,json]
+----
+{
+    "policies": [{
+      "name": "di",
+      "roles": ["data", "ingest" , "transform"],
+      "resources": {
+          "nodeCount": { "min": 3, "max": 8 },
+          "cpu": { "min": 2, "max": 8 },
+          "memory": { "min": "2Gi", "max": "16Gi" },
+          "storage": { "min": "64Gi", "max": "512Gi" }
+      }
+    },
+    {
+      "name": "ml",
+      "roles": ["ml"],
+      "resources": {
+          "nodeCount": { "min": 1, "max": 9 },
+          "cpu": { "min": 1, "max": 4 },
+          "memory": { "min": "2Gi", "max": "8Gi" }
+      }
+    }]
+}
+----
+
+WARNING: A node role should not be referenced in more than one autoscaling policy.
+
+In the case of storage the following restrictions apply:
+
+- Scaling the storage size automatically requires the `ExpandInUsePersistentVolumes` feature to be enabled. It also requires a storage class that supports link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion].
+- Only one persistent volume claim per Elasticsearch node is supported when autoscaling is enabled.
+- Volume size cannot be scaled down.
+
+[float]
+[id="{p}-{page_id}-resources"]
+=== Set the limits
+
+The value set for memory and CPU limits are computed by applying a ratio to the calculated resource request. For memory, the default ratio between the request and the limit is 1. This means that request and limit have the same value. For CPU the default ratio is 0, which means that no limit is set. You can change the default ratio between the request and the limit for both the CPU and memory ranges by using the `requestsToLimitsRatio` field.
+
+For example, you can remove the memory limit and set a CPU limit to twice the value of the request, as follows:
+
+[source,json]
+----
+{
+    "policies": [{
+        "name": "di-hot",
+        "roles": ["data_hot", "ingest", "transform"],
+        "resources": {
+            "nodeCount": {
+                "min": 2,
+                "max": 5
+            },
+            "cpu": {
+                "min": 1,
+                "max": 2,
+                "requestsToLimitsRatio": 2
+            },
+            "memory": {
+                "min": "2Gi",
+                "max": "6Gi",
+                "requestsToLimitsRatio": 0
+            }
+        }
+    }]
+}
+----
+
+You can find link:{eck_github}/blob/{eck_release_branch}/config/recipes/autoscaling/elasticsearch.yaml[a complete example in the ECK GitHub repository] which will also show you how to fine-tune the link:https://www.elastic.co/guide/en/elasticsearch/reference/current/autoscaling-deciders.html[autoscaling deciders].
+
+[float]
+[id="{p}-{page_id}-polling-interval"]
+=== Change the polling interval
+
+The Elasticsearch autoscaling capacity endpoint is polled every minute. This interval duration can be controlled using the `pollingPeriod` field in the autoscaling specification:
```

Comment:
It might be good to be more specific here, although it may seem obvious to us.
```suggestion
The Elasticsearch autoscaling capacity endpoint is polled every minute by the operator. This interval duration can be controlled using the `pollingPeriod` field in the autoscaling specification:
```